src/main/scala/nl.knaw.dans.easy.solr4files/ApplicationWiring.scala

```diff
@@ -15,15 +15,88 @@
  */
 package nl.knaw.dans.easy.solr4files
 
+import java.net.{ URI, URL }
+
+import nl.knaw.dans.easy.solr4files.components._
+import nl.knaw.dans.lib.error.{ CompositeException, TraversableTryExtensions }
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 
+import scala.util.{ Failure, Success, Try }
+
 /**
  * Initializes and wires together the components of this application.
  *
  * @param configuration the application configuration
  */
-class ApplicationWiring(configuration: Configuration) extends DebugEnhancedLogging
-// Mix in components
-{
-  // Configure components with configuration settings.
+class ApplicationWiring(configuration: Configuration)
+  extends DebugEnhancedLogging with VaultIO with Vault with Solr {
+
+  // don't need resolve for solr, URL gives more early errors TODO perhaps not enough early errors
+  override val solrUrl: URL = new URL(configuration.properties.getString("solr.url", ""))
+  override val vaultBaseUri: URI = new URI(configuration.properties.getString("vault.url", ""))
+
+  def initAllStores(): Try[FeedBackMessage] = {
+    getStoreNames
+      .flatMap(storeNames => updateStores(storeNames))
+      .map(results => s"Updated all bags of ${ results.size } stores ($vaultBaseUri)")
+  }
+
+  def initSingleStore(storeName: String): Try[FeedBackMessage] = {
+    getBagIds(storeName)
+      .flatMap(bagIds => updateBags(storeName, bagIds))
+      .map(results => s"Updated ${ results.size } bags of one store ($storeName)")
+  }
+
+  def update(storeName: String, bagId: String): Try[FeedBackMessage] = {
+    val bag = Bag(storeName, bagId, this)
+    for {
+      ddmXML <- bag.loadDDM
+      ddm = new DDM(ddmXML)
+      filesXML <- bag.loadFilesXML
+      files = (filesXML \ "file").map(FileItem(bag, ddm, _)).filter(f => !f.path.isEmpty && f.accessRights!="NONE")
+      _ <- deleteBag(bag.bagId)
+      feedbackMessage <- createDocs(bag, ddm, files)
+      _ <- commit()
+    } yield feedbackMessage
+  }
+
+  def delete(bagId: String): Try[FeedBackMessage] = for {
+    _ <- deleteBag(bagId)
+    _ <- commit()
+  } yield s"Deleted file documents for bag $bagId"
+
+  private def updateStores(storeNames: Seq[String]): Try[Seq[FeedBackMessage]] = {
+    storeNames
+      .map(initSingleStore)
+      .collectResults.recoverWith {
+      case t: CompositeException => throw new Exception(s"Tried to update ${ storeNames.size } stores, ${ t.getMessage() }", t)
```

Comment:
Make the exceptions as classes with nice names, e.g. `UpdateStoreException`.

```scala
case class UpdateStoreException(n: Int, msg: String, t: Throwable) extends Exception(s"Tried to update $n stores, $msg", t)
```

same for similar instances below